File: Python/1235.py
for i in range(int(input())):
    entrada = input()
    metade = int(len(entrada) / 2)

    parte1 = entrada[:metade]
    parte2 = entrada[metade:]

    novaParte1 = ""
    novaParte2 = ""

    for i in range(len(parte1) - 1, -1, -1):
        novaParte1 += parte1[i]
        novaParte2 += parte2[i]

    fraseCerta = novaParte1 + novaParte2

    print(fraseCerta)

File: C/1235.c
#include <stdio.h>
#include <string.h>

#define MAX_LEN 101

int main() {
    int N;
    scanf("%d", &N);
    
    while (getchar() != '\n');

    char entrada[MAX_LEN];

    for (int i = 0; i < N; i++) {
        fgets(entrada, MAX_LEN, stdin);
        entrada[strcspn(entrada, "\n")] = '\0';
        
        int len = strlen(entrada);
        int metade = len / 2;

        char parte1[MAX_LEN / 2 + 1];
        char parte2[MAX_LEN / 2 + 1];
        char novaParte1[MAX_LEN / 2 + 1];
        char novaParte2[MAX_LEN / 2 + 1];
        char fraseCerta[MAX_LEN];

        strncpy(parte1, entrada, metade);
        parte1[metade] = '\0';
        strncpy(parte2, entrada + metade, metade);
        parte2[metade] = '\0';

        for (int j = 0; j < metade; j++) {
            novaParte1[j] = parte1[metade - 1 - j];
            novaParte2[j] = parte2[metade - 1 - j];
        }
        novaParte1[metade] = '\0';
        novaParte2[metade] = '\0';

        snprintf(fraseCerta, sizeof(fraseCerta), "%s%s", novaParte1, novaParte2);

        printf("%s\n", fraseCerta);
    }

    return 0;
}
